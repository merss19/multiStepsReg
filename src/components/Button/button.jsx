"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var classNames = require("classnames");
var ButtonTypes;
(function (ButtonTypes) {
    ButtonTypes["info"] = "secondary";
    ButtonTypes["error"] = "error";
    ButtonTypes["success"] = "primary";
})(ButtonTypes = exports.ButtonTypes || (exports.ButtonTypes = {}));
exports.Button = function (_a) {
    var children = _a.children, onClick = _a.onClick, type = _a.type, styleBtn = _a.styleBtn, wide = _a.wide, prefix = _a.prefix;
    var classes = classNames(prefix, (_b = {},
        _b[prefix + "--" + styleBtn] = true,
        _b['wide'] = wide,
        _b));
    return (<button type={type} className={classes} onClick={onClick}>
      {children}
    </button>);
    var _b;
};
exports.Button.defaultProps = {
    styleBtn: ButtonTypes.info,
    type: 'button',
    wide: false,
    prefix: 'btn'
};
