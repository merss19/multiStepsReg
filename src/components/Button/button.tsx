import * as React from 'react';
import * as classNames from 'classnames';

export type ButtonType = ButtonTypes.info | ButtonTypes.error | ButtonTypes.success;

export enum ButtonTypes {
  info = 'secondary',
  error = 'error',
  success = 'primary'
}

interface OwnProps {
  onClick?: (data?: any) => any
  styleBtn?: ButtonType,
  type?: string,
  wide?: boolean
  prefix?: string
}

export const Button: React.SFC<OwnProps> = ({children, onClick, type, styleBtn, wide, prefix}) => {
  const classes = classNames(prefix, {
    [`${prefix}--${styleBtn}`]: true,
    ['wide']: wide
  });
  return (
    <button
      type={type}
      className={classes}
      onClick={onClick}
    >
      {children}
    </button>)
};

Button.defaultProps = {
  styleBtn: ButtonTypes.info,
  type: 'button',
  wide: false,
  prefix: 'btn'
};