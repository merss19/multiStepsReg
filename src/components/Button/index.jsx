"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var button_1 = require("./button");
exports.Button = button_1.Button;
exports.ButtonTypes = button_1.ButtonTypes;
