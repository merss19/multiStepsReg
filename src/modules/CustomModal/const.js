"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var ModalTypes;
(function (ModalTypes) {
    ModalTypes["info"] = "info";
    ModalTypes["error"] = "error";
    ModalTypes["success"] = "success";
    ModalTypes["loading"] = "loading";
})(ModalTypes = exports.ModalTypes || (exports.ModalTypes = {}));
