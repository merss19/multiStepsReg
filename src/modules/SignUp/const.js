"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var Steps;
(function (Steps) {
    Steps[Steps["one"] = 1] = "one";
    Steps[Steps["two"] = 2] = "two";
    Steps[Steps["three"] = 3] = "three";
    Steps[Steps["four"] = 4] = "four";
})(Steps = exports.Steps || (exports.Steps = {}));
;
var Programs;
(function (Programs) {
    Programs["hero"] = "hero";
    Programs["mother"] = "mother";
    Programs["extreme"] = "extreme";
    Programs["tomorrow"] = "tomorrow";
})(Programs = exports.Programs || (exports.Programs = {}));
;
var ProgramsName;
(function (ProgramsName) {
    ProgramsName["hero"] = "#\u042F\u0433\u0435\u0440\u043E\u0439";
    ProgramsName["mother"] = "#\u041C\u0430\u043C\u0430 \u043C\u043E\u0436\u0435\u0442";
    ProgramsName["extreme"] = "#\u042D\u043A\u0441\u0442\u0440\u0438\u043C";
    ProgramsName["tomorrow"] = "#\u0417\u0430\u0432\u0442\u0440\u0430";
})(ProgramsName = exports.ProgramsName || (exports.ProgramsName = {}));
;
var ProgramNum;
(function (ProgramNum) {
    ProgramNum[ProgramNum["hero"] = 1] = "hero";
    ProgramNum[ProgramNum["mother"] = 2] = "mother";
    ProgramNum[ProgramNum["extreme"] = 3] = "extreme";
    ProgramNum[ProgramNum["tomorrow"] = 0] = "tomorrow";
    ProgramNum[ProgramNum["count"] = 4] = "count";
})(ProgramNum = exports.ProgramNum || (exports.ProgramNum = {}));
;
