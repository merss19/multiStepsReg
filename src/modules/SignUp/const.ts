export enum Steps {
  one = 1,
  two = 2,
  three = 3,
  four = 4
};

export enum Programs {
  hero = 'hero',
  mother = 'mother',
  extreme = 'extreme',
  tomorrow = 'tomorrow'
};

export enum ProgramsName {
  hero = '#Ягерой',
  mother = '#Мама может',
  extreme = '#Экстрим',
  tomorrow = '#Завтра'
};

export enum ProgramNum {
  hero = 1,
  mother = 2,
  extreme = 3,
  tomorrow = 0,
  count = 4
};