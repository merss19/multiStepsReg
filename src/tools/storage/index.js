"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var cookie = require("react-cookie");
exports.storage = {
    save: function (name, data, path) {
        cookie.save(name, data, path);
    },
    load: function (name) {
        return cookie.load(name);
    },
    remove: function (name, path) {
        return cookie.remove(name, path);
    },
    loadState: function (name) {
        try {
            var serializedState = localStorage.getItem(name);
            if (!serializedState) {
                return undefined;
            }
            return JSON.parse(serializedState);
        }
        catch (err) {
            return undefined;
        }
    },
    saveState: function (data, name) {
        try {
            var serializedState = JSON.stringify(data);
            localStorage.setItem(name, serializedState);
        }
        catch (err) {
            //console.log(err)
        }
    },
    removeState: function (name) {
        try {
            localStorage.removeItem(name);
        }
        catch (err) {
            ///console.log(err)
        }
    }
};
