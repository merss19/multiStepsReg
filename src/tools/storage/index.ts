import * as cookie from 'react-cookie';

interface StorageAction {
	save: (name: string, data: any, path?: object) => void
	load: (name: string) => any
	remove: (name: string, path?: object) => void
	loadState: (name: string) => any,
	saveState: (data: any, name: string) => void
	removeState: (name: string) => void
}
export const storage: StorageAction = {
  save: (name: string, data: any, path?: object): void => {
    cookie.save(name, data, path)
  },
  load: (name: string): any => {
    return cookie.load(name)
  },
  remove: (name: string, path?: object): any => {
    return cookie.remove(name, path)
  },
  loadState: (name: string) => {
    try {
      const serializedState = localStorage.getItem(name);
      if (!serializedState) {
        return undefined
      }
      return JSON.parse(serializedState);
    } catch (err) {
      return undefined
    }
  },
  saveState: (data: any, name: string) => {
    try {
      const serializedState = JSON.stringify(data);
      localStorage.setItem(name, serializedState)
    } catch (err) {
      //console.log(err)
    }
  },
  removeState: (name: string) => {
    try {
      localStorage.removeItem(name)
    } catch (err) {
      ///console.log(err)
    }
  }
};