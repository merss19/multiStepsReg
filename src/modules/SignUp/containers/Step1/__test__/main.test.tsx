import {shallow, mount} from 'enzyme';
import StepOneReduxForm, {StepOneComponent, StepOne} from '../main';
import * as React from 'react';
import toJson from 'enzyme-to-json';
import * as sinon from 'sinon';
import {Provider} from 'react-redux';
import configureStore from '../../../../../store';
import {BrowserRouter as Router} from 'react-router-dom';
import {configure} from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import {storage} from '../../../../../tools/storage/index';
import {Steps} from '../../../const';
import {Button, ButtonTypes} from "../../../../../components/Button/button";
import * as Immutable from 'immutable';

const store = configureStore();
configure({adapter: new Adapter()});
let data = [
  {
    id: 1,
    name: 'hero',
    isActive: true
  },
  {
    id: 2,
    name: 'mother',
    isActive: false
  },
  {
    id: 3,
    name: 'extreme',
    isActive: false
  },
  {
    id: 4,
    name: 'tomorrow',
    isActive: false
  }
];

let changeStep,
  getPrograms,
  userCreate,
  setProgramName,
  handleSubmit,
  setToken,
  toggleModal,
  setError,
  isFetching,
  isLoad,
  isError,
  step,
  errMsg,
  signUpData,
  dataRes,
  match,
  location,
  history;
const start = () => {
  beforeEach(() => {
    changeStep = jest.fn(),
      getPrograms = jest.fn(),
      userCreate = jest.fn((data, callback) => callback()),
      setProgramName = jest.fn(),
      setToken = jest.fn(),
      handleSubmit = jest.fn((callback) => callback()),
      toggleModal = jest.fn(),
      setError = jest.fn(),
      isFetching = false,
      isLoad = false,
      isError = false,
      errMsg = 'error',
      step = Steps.one,
      dataRes = {
        authToken: 'qweqwe',
        promoName: 'tele2'
      },
      signUpData = {
        email: 'mail@mail.ru',
        password: 'qweqwe',
        passwordAgain: 'qweqwe',
        gender: 'male',
      },
      match = {
        isExact: true,
        params: {},
        path: "/task",
        url: "/task"
      },
      location = {},
      history = {
        replace: jest.fn()
      }
  });
  const props = {
    getPrograms: getPrograms,
    userCreate: userCreate,
    handleSubmit: handleSubmit,
    changeStep: changeStep,
    setProgramName: setProgramName,
    setToken: setToken,
    toggleModal: toggleModal,
    setError: setError,
    step: step,
    errMsg: errMsg,
    isFetching: isFetching,
    isLoad: isLoad,
    isError: isError,
    match: match,
    location: location,
    history: history
  };
  const buildComponent = () => {
    return shallow(<StepOneComponent {...props}/>)
  };
  const buildComponentMount = () => {
    return mount(
      <Provider store={store}>
        <Router>
          <StepOneReduxForm {...props}/>
        </Router>
      </Provider>)
  };
  return {
    buildComponentMount,
    props,
    buildComponent
  }
};
describe('Component StepOne', () => {
  let component;
  start();
  beforeEach(() => {
    component = start().buildComponent();
  });
  it('renders component', () => {
    expect(component).toHaveLength(1);
  });
  it('renders component Mount', () => {
    const component = start().buildComponentMount();
    expect(component).toHaveLength(1);
  });
  it('renders correctly', () => {
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });
  describe('Test shouldComponentUpdate', () => {
    it('step', () => {
      const instance = component.instance() as StepOneComponent;
      step = Steps.two;
      let props = {
        ...start().props,
        step: step
      };
      const shouldUpdate = instance.shouldComponentUpdate(props);
      expect(shouldUpdate).toBe(true)
    });
 /*   it('isFetching', () => {
      const instance = component.instance() as StepOneComponent;
      isFetching = true;
      let props = {
        ...start().props,
        isFetching: isFetching
      };
      const shouldUpdate = instance.shouldComponentUpdate(props);
      expect(shouldUpdate).toBe(true)
    });
    it('isError', () => {
      const instance = component.instance() as StepOneComponent;
      isError = true;
      let props = {
        ...start().props,
        isError: isError
      };
      const shouldUpdate = instance.shouldComponentUpdate(props);
      expect(shouldUpdate).toBe(true)
    });
    it('errMsg', () => {
      const instance = component.instance() as StepOneComponent;
      errMsg = 'test';
      let props = {
        ...start().props,
        errMsg: errMsg
      };
      const shouldUpdate = instance.shouldComponentUpdate(props);
      expect(shouldUpdate).toBe(true)
    });*/
  });
  it('call onSubmit', () => {
    const spy = sinon.spy(StepOneComponent.prototype, 'resultAction');
    const component = start().buildComponent();
    const instance = component.instance() as StepOneComponent;
    instance.onSubmit(signUpData);
    const cb = jest.fn(() => true);
    expect(userCreate).toHaveBeenCalled();
    component.instance().props.userCreate(signUpData, cb);
    expect(spy.called).toBe(true);
  });
  it('call resultAction, dataRes.authToken !== null', () => {
    const instance = component.instance() as StepOneComponent;
    instance.resultAction(dataRes);
    expect(setToken).toHaveBeenCalledWith(dataRes.authToken);
    expect(changeStep).toHaveBeenCalledWith(Steps.two);
    expect(storage.load('token')).toBe(dataRes.authToken);
    expect(storage.load('promoName')).toBe(dataRes.promoName);
    storage.remove('token');
    storage.remove('promoName');
  });
  it('call resultAction, dataRes.authToken === null', () => {
    dataRes = {
      authToken: null,
      promoName: 'tele2'
    };
    const instance = component.instance() as StepOneComponent;
    instance.resultAction(dataRes);
    expect(setToken).not.toHaveBeenCalled();
    expect(changeStep).not.toHaveBeenCalled();
    expect(storage.load('token')).toBeUndefined();
    expect(storage.load('promoName')).toBeUndefined();
  });
  describe('call render()', () => {
    it('call renderHeader', () => {
      const spy = sinon.spy(StepOneComponent.prototype, 'renderHeader');
      const component = start().buildComponent();
      expect(spy.called).toBe(true);
    });
    it('render button', () => {
      const component = start().buildComponent();
      const button = component.find(Button)
      expect(button.exists()).toBe(true);
      expect(button.props().type).toBe('submit');
      expect(button.props().styleBtn).toBe(ButtonTypes.info);
      expect(button.props().wide).toBe(true);
    });
    it('render Fields', () => {
      const component = start().buildComponent();
      const email = component.find('Field[name="email"]');
      const password = component.find('Field[name="password"]');
      const passwordAgain = component.find('Field[name="passwordAgain"]');
      const gender = component.find('Field[name="gender"]');
      const accept = component.find('Field[name="accept"]');
      expect(email.exists()).toBe(true);
      expect(password.exists()).toBe(true);
      expect(accept.exists()).toBe(true);
      expect(passwordAgain.exists()).toBe(true);
      expect(gender).toHaveLength(2);
    });
    it('click Button', () => {
      const component = start().buildComponent();
      component.find('form').simulate('submit')
      expect(component.instance().props.handleSubmit).toHaveBeenCalled()
    });
  });
});


