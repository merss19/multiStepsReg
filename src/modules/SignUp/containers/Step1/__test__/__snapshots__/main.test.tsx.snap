// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component StepOne renders correctly 1`] = `
<div>
  <div
    className="entry__header"
  >
    <h2
      className="entry__title entry__title--auth text-center "
    >
      Регистрация
    </h2>
  </div>
  <form
    className="entry__box"
  >
    <Field
      component={[Function]}
      name="email"
      placeholder="Ваш e-mail"
      type="text"
    />
    <Field
      component={[Function]}
      name="password"
      placeholder="Ваш пароль"
      type="password"
    />
    <Field
      component={[Function]}
      name="passwordAgain"
      placeholder="Пароль повторно"
      type="password"
    />
    <div
      className="gender"
    >
      <div
        className="gender__group"
      >
        <p
          className="gender__title"
        >
          Ваш пол:
        </p>
        <Field
          component={[Function]}
          id="male"
          name="gender"
          title="Мужчина"
          type="radio"
          validate={[Function]}
          value="male"
        />
        <Field
          component={[Function]}
          id="female"
          name="gender"
          title="Женщина"
          type="radio"
          validate={[Function]}
          value="female"
        />
      </div>
      <hr
        className="gender__hr"
      />
    </div>
    <div
      className="checkboxes"
    >
      <Field
        component={[Function]}
        id="accept"
        name="accept"
        title="Принять условия "
      />
    </div>
    <Component
      prefix="btn"
      styleBtn="secondary"
      type="submit"
      wide={true}
    >
      Зарегистрироваться
    </Component>
    <div
      className="entry__link text-center mtb20"
    >
      <div
        className="entry-nav__item"
      >
        <Link
          replace={false}
          to="/"
        >
          Войти
        </Link>
      </div>
      <div
        className="entry-nav__item"
      >
        <Link
          replace={false}
          to="/restore"
        >
          Забыли пароль?
        </Link>
      </div>
    </div>
  </form>
</div>
`;
